x-pack/test/apm_api_integration/tests/span_links/linked_parents.spec.ts

```diff
@@ -0,0 +1,148 @@
+    () => {
+      let externalSpanLinks: SpanLink[];
+      let serviceBTraceId: string;
+      let serviceBTransactionId: string;
+      before(async () => {
+        const serviceBAsArray = generateIncomeEventsSpanLinks().toArray();
+        const transaction = serviceBAsArray.slice(0, 1)[0];
+        serviceBTraceId = transaction['trace.id']!;
+        serviceBTransactionId = transaction['transaction.id']!;
+
+        externalSpanLinks = generateExternalSpanLinks(3);
+
+        const instanceJava = apm.service('service-A', 'production', 'java').instance('instance-a');
+        const serviceAEvents = timerange(start, end)
+          .interval('1m')
+          .rate(1)
+          .generator((timestamp) => {
+            return instanceJava
+              .transaction('GET /service_A')
+              .defaults({
+                'span.links': [...externalSpanLinks, ...getSpanLinksFromEvents(serviceBAsArray)],
+              })
+              .timestamp(timestamp)
+              .duration(1000)
+              .success()
+              .children(
+                instanceJava
+                  .span('get_service_A', 'db', 'elasticsearch')
+                  .timestamp(timestamp + 50)
+                  .duration(900)
+                  .success()
+              );
+          });
+
+        await synthtraceEsClient.index(
+          new EntityArrayIterable(serviceBAsArray).merge(serviceAEvents)
+        );
+      });
+
+      after(() => synthtraceEsClient.clean());
+
+      describe('should have span links details', async () => {
+        let spanLinksDetails: Awaited<ReturnType<typeof getIncomingSpanDetails>>['body'];
+        before(async () => {
+          const response = await getIncomingSpanDetails({
+            kuery: '',
+            traceId: serviceBTraceId,
+            spanId: serviceBTransactionId,
+          });
+          expect(response.status).to.eql(200);
+          spanLinksDetails = response.body;
+        });
+
+        it('returns service-A as outgoing span links', () => {
+          expect(
+            spanLinksDetails.spanLinksDetails.filter(
+              (item) => item.details?.serviceName === 'service-A'
+            ).length
+          ).to.be.greaterThan(0);
+        });
+      });
+
+      describe('with kuery', async () => {
+        let spanLinksDetails: Awaited<ReturnType<typeof getIncomingSpanDetails>>['body'];
+        before(async () => {
+          const response = await getIncomingSpanDetails({
+            traceId: serviceBTraceId,
+            spanId: serviceBTransactionId,
+            kuery: 'service-A',
+          });
+          expect(response.status).to.eql(200);
+          spanLinksDetails = response.body;
+        });
+
+        it('does not return external links', () => {
+          expect(
+            spanLinksDetails.spanLinksDetails.map((item) => `${item.traceId}:${item.spanId}`)
+          ).not.to.be.eql(externalSpanLinks.map((item) => `${item.trace?.id}:${item.span.id}`));
```

Comment:
I'm not sure if this is a useful test. E.g., you're not sorting the items, so if the order is different, but external links are still returned, this test will pass.

Reply:
new tests here https://github.com/elastic/kibana/pull/126630/commits/651bf45f68ea0236a889afeb760d73586821374d#diff-5a56fd212eecdb6f4035475e1af2689b8e5f99ee97b8073f08964356ca0a1555R12